platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/SyncGossip.java

```diff
@@ -328,6 +329,44 @@ private void buildSyncProtocolThreads(
             final ProtocolConfig protocolConfig,
             final ReconnectConfig reconnectConfig,
             final EventConfig eventConfig) {
+        final ProtocolFactory syncProtocolFactory = new SyncProtocolFactory(
```

Comment:
I think it would be good that we immediately pull out the construction of this out of this class, and just pass in a list of factories. Then we are really making a move towards decoupling the protocols from the network stack.

Reply:
Agreed, but i don't think we should immediately do this. On quick glance, it will take a lot of heavy lifting of a bunch of stuff out of SyncGossip to enable external creation of the protocol factories. Not impossible, but a major refactor of syncgossip should belong in a separate, planned work.